### ISSUE TITLE
Wrong result when entity has null relationship and is not the first one in array

### ISSUE THREAD

**Issue by olOwOlo** [olosegres/jsona#17]
"jsona": "^1.1.10"

---

```js
deserialize({
  "data": [
    {
      "type": "category",
      "id": "3",
      "attributes": {
        "slug": "ya"
      },
      "relationships": {
        "parent": {
          "data": {
            "type": "category",
            "id": "0"
          }
        }
      }
    },
    {
      "type": "category",
      "id": "0",
      "attributes": {
        "slug": "home"
      },
      "relationships": {
        "parent": {
          "data": null
        }
      }
    }
  ]
})
```

output

```json
[
    {
        "type": "category", 
        "id": "3", 
        "slug": "ya", 
        "parent": {
            "type": "category", 
            "id": "0"
        }, 
        "relationshipNames": [
            "parent"
        ]
    }, 
    {
        "type": "category", 
        "id": "0"
    }
]
```

However, when it is the first element in the array,

```js
deserialize({
    "data": [
        {
            "type": "category", 
            "id": "0", 
            "attributes": {
                "slug": "home"
            }, 
            "relationships": {
                "parent": {
                    "data": null
                }
            }
        }
    ]
})
```

it works good.

```json
[
    {
        "type": "category", 
        "id": "0", 
        "slug": "home"
    }
]
```

---

In addition, I think it would be better to keep the null value, like this:
```json
[
    {
        "type": "category", 
        "id": "0", 
        "slug": "home",
        "parent": null, 
        "relationshipNames": [
            "parent"
        ]
    }
]
```

**Comment by olosegres** [comment]
~olOwOlo, hi
Sorry for the long wait, bug fixed in v1.1.11

Thank you

**Comment by olOwOlo** [comment]
It works now! Thank you very much!
